src/lib/rest/StringFilter.cpp

```diff
@@ -1187,8 +1294,76 @@ bool StringFilter::match(ContextElementResponse* cerP)
         return false;
       }
       break;
+
+    case SfopMatchPattern:
+      return itemP->matchPattern(caP);
+      break;
     }
   }
 
   return true;
 }
+
+
+
+/* ****************************************************************************
+*
+* StringFilter::clone - 
+*/
+StringFilter* StringFilter::clone(std::string* errorStringP)
+{
+  StringFilter* sfP = new StringFilter();
+
+  for (unsigned int ix = 0; ix < filters.size(); ++ix)
+  {
+    StringFilterItem  sfi;
+
+    if (!sfi.fill(&filters[ix], errorStringP))
+    {
+      delete sfP;
+      return NULL;
+    }
+
+    sfP->filters.push_back(sfi);
+
+    // Next line is to avoid double-free at StringFilterItem destructor
```

Comment:
I like the wording you use for an analogous situation at https://github.com/telefonicaid/fiware-orion/pull/2081/files#diff-ef659d8f6e392bb5d95fdfbcd0997815R938. Please, replace my original comment in that part wiith yours.